src/protocols/privateChannel.ts

```diff
@@ -0,0 +1,188 @@
+import { PrivateCommand, calculatePrivateTopic, PrivateCommandPost, PrivateCommandRemove } from './privateSharing';
+import { ChapterReference, Timeline, PartialChapter, readTimeline, makePartialChapter, fetchTimeline, getNewestChapterId, uploadTimeline } from './timeline';
+import { MutualContact } from '../models/Contact';
+import { ProtocolStorage } from './ProtocolStorage';
+import { HexString } from '../helpers/opaqueTypes';
+import { serialize, deserialize } from '../social/serialization';
+import { stringToUint8Array, hexToUint8Array, Uint8ArrayToString } from '../helpers/conversion';
+import { ProtocolCrypto } from './ProtocolCrypto';
+import { PrivatePost, PostWithId } from '../models/Post';
+import { Debug } from '../Debug';
+
+export interface PrivateChannel {
+    unsyncedCommands: PrivateCommand[];
+    lastSyncedChapterId: ChapterReference | undefined;
+    peerLastSeenChapterId: ChapterReference | undefined;
+}
+
+export const makeEmptyPrivateChannel = (): PrivateChannel => ({
+    unsyncedCommands: [],
+    lastSyncedChapterId: undefined,
+    peerLastSeenChapterId: undefined,
+});
+
+interface PrivateChannelUpdate {
+    topic: HexString;
+    privateChannel: PrivateChannel;
+    syncedLocalTimeline: Timeline<PrivateCommand>;
+    peerTimeline: Timeline<PrivateCommand>;
+}
+
+const privateChannelAppendCommand = (privateChannel: PrivateChannel, command: PrivateCommand): PrivateChannel => {
+    return {
+        ...privateChannel,
+        unsyncedCommands: [command, ...privateChannel.unsyncedCommands],
+    };
+};
+
+export const privateChannelAddPost = (privateChannel: PrivateChannel, post: PostWithId): PrivateChannel => {
+    const command: PrivateCommandPost = {
+        type: 'post',
+        version: 1,
+        protocol: 'private',
+        post,
+    };
+    return privateChannelAppendCommand(privateChannel, command);
+};
+
+export const privateChannelRemovePost = (privateChannel: PrivateChannel, id: HexString): PrivateChannel => {
+    const command: PrivateCommandRemove = {
+        type: 'remove',
+        version: 1,
+        protocol: 'private',
+        id,
+    };
+    return privateChannelAppendCommand(privateChannel, command);
+};
+
+export const uploadUnsyncedTimeline = async (
+    unsyncedTimeline: Timeline<PrivateCommand>,
+    lastSyncedChapterId: ChapterReference | undefined,
+    address: HexString,
+    topic: HexString,
+    sharedSecret: HexString,
+    storage: ProtocolStorage,
```

Comment:
Protocol describes how data is stored on Swarm. They could be also called APIs, but since on Swarm there is no difference between accessing the data compared to how it stored, I prefer to use Protocol.
We also have the (now obsolete) Social API which can be considered a protocol in this terminology. Also we use the Timeline protocol as a lower-level building block for other protocols.

Reply:
It's clear what `a protocol` is, but calling it `Protocol` is not very informative.